wandb/sdk/lib/filenames.py

```diff
@@ -32,7 +32,10 @@ def filtered_dir(
     root: str, include_fn: Callable[[str], bool], exclude_fn: Callable[[str], bool]
 ) -> Generator[str, None, None]:
     """Simple generator to walk a directory"""
-    for dirpath, _, files in os.walk(root):
+    for dirpath, dirs, files in os.walk(root):
+        for dir in dirs:
+            if exclude_fn(dir):
```

Comment:
your logic doesn't handle the case where `include_fn` and `exclude_fn` disagree. 
This should probably be resolved and communicated to the user if there are some contradictions. 

Also, would mind adding tests for this change?

Reply:
But also when such an event happens on the file level (i.e. when a file is both included and occluded) there is no communication to the user and the file is being excluded.. I expect the same to happen on the folder level.
So what you are describing is another feature.. 
Regarding testing, is there an existing test for this function I can expand?

Reply:
Well, like you mentioned in the file case we check both `include_fn` and `exclude_fn` so for the extended logic to be complete you need to handle this case as well. Potentially, you could modify your condition to account for these cases. 

Regarding testing, there is no specific test for this function, but you could use one of the test that test the `log_code` function since it uses this function, but I think it would be better to test this function separate as a unit test.